### PR TITLE
Add core WG notes for 2021-05-21.

### PR DESCRIPTION
**Pull request by jrvanwhy**
[Rendered](https://github.com/jrvanwhy/tock/blob/core-notes-2021-05-21/doc/wg/core/notes/core-notes-2021-05-21.md)

### CODE REVIEW

File: doc/wg/core/notes/core-notes-2021-05-21.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+## Updates
+
+### Allow buffer overlap handling
+ * Phil: I sent out the latest numbers I have for doing runtime checks for
+   buffer overlap. Given the [discussions on
+   tock-dev](https://groups.google.com/g/tock-dev/c/cY0-eKc6aos), we can figure
+   out whether they are necessary or desirable. Basic summary is they're in the
+   range of 70 - 300 cycles depending on the number of outstanding buffers. I'm
+   only looking at 0-8 outstanding buffers. Most of the overhead, looking at the
+   instructions, is register spillage. For example, the "insert zero" caseis
```
Comment (brghena) [inline doc/wg/core/notes/core-notes-2021-05-21.md:24]:
```suggestion
   instructions, is register spillage. For example, the "insert zero" case is
```

File: doc/wg/core/notes/core-notes-2021-05-21.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+   lightweight contexts seems to be high-overhead for our scenario.
+
+## Process Console Extension presentation
+ * Phil: To introduce Andrew, he is working on his senior project for some fun
+   implementation systems work. We're talking and Brad mentioned that improving
+   the process console and making it more featureful is useful. Andrew has been
+   working on it the past 8-ish weeks. He'll talk about what he's done and some
+   small implications to the kernel.
+ * Andrew: Process console started without a true writer, it used the debug
+   writer. It used its own liter just to add new lines at the end of user input.
```
Comment (brghena) [inline doc/wg/core/notes/core-notes-2021-05-21.md:85]:
```suggestion
   writer. It used its own writer just to add new lines at the end of user input.
```

I think this is the fix?

File: doc/wg/core/notes/core-notes-2021-05-21.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+   otherwise needed a large queue, instead we use a state machine to allow the
+   queue to be small without dropping packets.
+ * Andrew: Added a command to print the memory map for a process. This is a
+   large print so we had to use a state machine to print out several parts
+   rather than one large chunk.
+ * Andrew: Added a kernel command that has more of an effect on the kernel,
+   mostly to get information to print things out. Added the ability to print the
+   kernel map of the kernel, and a macro to see which drivers are available in
+   the board structure. Could not think of any approaches other than a macro in
+   the board's mail file, so we decided to use a macro. Macro seems to work well
```
Comment (brghena) [inline doc/wg/core/notes/core-notes-2021-05-21.md:98]:
```suggestion
   the board's main file, so we decided to use a macro. Macro seems to work well
```

**Review by brghena — COMMENTED** [review]
Thanks for writing this up! Really helpful for catching up.

File: doc/wg/core/notes/core-notes-2021-05-21.md (hunk 1 above)
Comment (jrvanwhy) [reply doc/wg/core/notes/core-notes-2021-05-21.md]: Fixed.

File: doc/wg/core/notes/core-notes-2021-05-21.md (hunk 3 above)
Comment (jrvanwhy) [reply doc/wg/core/notes/core-notes-2021-05-21.md]: Fixed, thanks.

File: doc/wg/core/notes/core-notes-2021-05-21.md (hunk 2 above)
Comment (jrvanwhy) [reply doc/wg/core/notes/core-notes-2021-05-21.md]: That's correct. Fixed.

**Review by hudson-ayers — APPROVED** [review]

**Pull request merged**